js/web/components/CollectorModal.js

```diff
@@ -0,0 +1,154 @@
+    if (wallet?.publicKey) {
+      if (collection[releasePubkey] > 0) {
+        collectorsList.push(wallet.publicKey)
+      } else if (collectorsList.includes(wallet.publicKey)) {
+        collectorsList.remove(wallet.publicKey)
+      }
+    }
+    setCollectors(collectorsList)
+  }
+
+  return (
+    <Box color={'wj'}>
+      <Cta onClick={() => setOpen(true)} variant="body2" align="left" paddingBottom="10px">
+        {`View Collectors ${collectors ? `(${collectors.length})` : ''}`}
+      </Cta>
+      <StyledModal
+        aria-labelledby="transition-modal-title"
+        aria-describedby="transition-modal-description"
+        className={classes.modal}
+        open={open}
+        onClose={() => setOpen(false)}
+        closeAfterTransition
+        BackdropComponent={Backdrop}
+        BackdropProps={{
+          timeout: 500,
+        }}
+      >
+        <StyledPaper>
+          <Header>
+            <Typography fontWeight="700">{`${metadata.properties.artist.substring(0, 100)} - "${metadata.properties.title.substring(0, 100)}" Collectors`}</Typography>
+          </Header>
+          <HistoryTable>
+            <TableBody>
+              {collectors &&
+                collectors.map((entry, i) => {
+                  return (
+                    <tr key={i}>
+                      <td>
+                        <Link href={`/collection/${entry}`} passHref>
+                          {`${entry} (View Collection)`}
+                        </Link>
+                      </td>
+                    </tr>
+                  )
+                })}
+            </TableBody>
+          </HistoryTable>
+        </StyledPaper>
+      </StyledModal>
+    </Box>
+  )
+}
+
+const PREFIX = 'ExchangeHistoryModal'
+
+const classes = {
+  exchangeHistoryCta: `${PREFIX}-exchangeHistoryCta`,
+  modal: `${PREFIX}-modal`,
+  paper: `${PREFIX}-paper`,
+  header: `${PREFIX}-header`,
+  historyTable: `${PREFIX}-historyTable`,
+  historyTableBody: `${PREFIX}-historyTableBody`,
+}
+
+const Cta = styled(Typography)(({ theme }) => ({
+  cursor: 'pointer',
+  '& span': {
+    color: `${theme.palette.blue}`,
+  },
+  ':hover': {
+    opacity: 0.5
+  }
+}))
+
+const StyledModal = styled(Modal)(() => ({
+  display: 'flex',
+  alignItems: 'center',
+  justifyContent: 'center',
+  'a:hover': {
+    opacity: 0.5
+  }
+}))
+
+const StyledPaper = styled(Paper)(({ theme }) => ({
+  backgroundColor: theme.palette.background.paper,
+  boxShadow: theme.shadows[5],
+  padding: theme.spacing(6, 4),
+  ...theme.gradient,
+  zIndex: '10',
+}))
+
+const Header = styled(Typography)(({ theme }) => ({
+  fontSize: '26px',
+  textAlign: 'center',
+  display: 'flex',
+  flexDirection: 'column',
+  fontWeight: '700',
+  lineHeight: '29.9px',
+  color: theme.palette.white,
+}))
+
+const HistoryTable = styled('table')(({ theme }) => ({
+  padding: `${theme.spacing(1, 1)}`,
+  display: 'block',
+  maxHeight: '50vh',
+  overflow: 'scroll',
```

Comment:
we should just apply `pverflowY` + `HistoryTable` should be renamed to `Collector` namespace